gateway/handler_car.go

```diff
@@ -30,18 +37,52 @@ func (i *handler) serveCAR(ctx context.Context, w http.ResponseWriter, r *http.R
 		return false
 	}
 
-	pathMetadata, carFile, errCh, err := i.api.GetCAR(ctx, imPath)
+	queryParams := r.URL.Query()
+	rangeStr, hasRange := queryParams.Get(carRangeBytesKey), queryParams.Has(carRangeBytesKey)
+	scopeStr, hasScope := queryParams.Get(carTerminalElementTypeKey), queryParams.Has(carTerminalElementTypeKey)
+
+	params := CarParams{}
+	if hasRange {
+		rng, err := rangeStrToByteRange(rangeStr)
+		if err != nil {
+			webError(w, err, http.StatusBadRequest)
+			return false
+		}
+		params.Range = &rng
+	}
+
+	if hasScope {
+		switch s := DagScope(scopeStr); s {
+		case dagScopeEntity, dagScopeAll, dagScopeBlock:
+			params.Scope = s
+		default:
+			err := fmt.Errorf("unsupported dag-scope %s", scopeStr)
```

Comment:
in case the client  got too creative
```suggestion
			err := fmt.Errorf("unsupported dag-scope %q", scopeStr)
```